### ISSUE TITLE
Unable to reproduce the domainbed accuracies

### ISSUE THREAD

**Issue by Harsh-Sensei** [Luodian/Generalizable-Mixture-of-Experts#4]
Hello!
I am facing trouble reproducing the accuracies reported in paper(https://arxiv.org/pdf/2206.04046v5.pdf). I used the default hyperparameters mentioned in the paper, and obtained the following results:
1. VLCS : 78.8(Obtained); 80.2(Reported in paper)
2. PACS : 86.9(Obtained); 88.1(Reported in paper)
3. OfficeHome : 72.7(Obtained); 74.2(Reported in paper)
Can you please help me with this issue? Am I missing any implementation details?

**Comment by Luodian** [comment]
For those three datasets, I can paste the exact hparams used to reproduce these results.

```sh
HParams:
	batch_size: 32
	class_balanced: False
	data_augmentation: True
	freeze_bn: True
	lr: 3e-05
	mlp_depth: 3
	mlp_dropout: 0.0
	nonlinear_classifier: False
	optimizer: adam
	pretrained: True
	resnet18: False
	resnet_dropout: 0.1
	val_augment: False
	weight_decay: 1e-06
HParams:
	batch_size: 32
	class_balanced: False
	data_augmentation: True
	freeze_bn: True
	lr: 3e-05
	mlp_depth: 3
	mlp_dropout: 0.0
	nonlinear_classifier: False
	optimizer: adam
	pretrained: True
	resnet18: False
	resnet_dropout: 0.1
	val_augment: False
	weight_decay: 1e-06
```
The results.txt is 

```sh
-------- Dataset: PACS, model selection method: training-domain validation set
Algorithm             A                     C                     P                     S                     Avg                  
SFMOE                 89.9 +/- 0.0          82.1 +/- 0.0          99.2 +/- 0.0          81.2 +/- 0.0          88.1
```

**Comment by ghost** [comment]
Hello!
I noticed that the bias of Avg is +-0, How many seeds did you run? 1 seed or other seed? And could you please share me the setting of terra?

**Comment by Luodian** [comment]
> Hello! I noticed that the bias of Avg is +-0, How many seeds did you run? 1 seed or other seed? And could you please share me the setting of terra?

Hi! I think at that time, I only ran with 1 seed to quickly provide the reproduced results. I need sometime to find the Terra's configs since it's been a long time, I will get you back as soon as possible!

**Comment by ghost** [comment]
Thank you for your reply! I have done some GMoE multi-seed experiments and often found that acc decreases as more seeds are added, could you teach me how to maintain a high acc even with multiple seeds? Or could you please tell me the specific muti-seed that you used in experiments?
